system/jlib/jthread.cpp

```diff
@@ -1048,16 +1049,15 @@ class CThreadPool: public CThreadPoolBase, implements IThreadPool, public CInter
                     overAllTimer.reset();
                 }
             }
-            CPooledThreadWrapper &t = allocThread();
-            if (name)
-                t.setName(name);
-            if (inheritThreadContext)
-                t.captureThreadLoggingInfo();
-            t.go(param);
-            ret = t.queryHandle();
+            t = &allocThread();
         }
-        Sleep(0);
-        return ret;
+        if (name)
+            t->setName(name);
+        if (inheritThreadContext)
+            t->captureThreadLoggingInfo();
+        t->go(param);
+        Sleep(0); // Potentially yield so that the new thread has a chance to start processing.
```

Comment:
@mckellyln do you think this is still a good idea?

Reply:
Good question, probably not.
Maybe if we are oversubscribed then yes.